src/main/java/seedu/address/model/person/Block.java

```diff
@@ -0,0 +1,87 @@
+package seedu.address.model.person;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * Represents a Person's block in the address book.
+ * Guarantees: immutable; is valid as declared in {@link #isValidBlock(String)}
+ */
+public class Block {
+
+    public static final String MESSAGE_CONSTRAINTS =
+            "Only the following blocks exists: " + getBlockEnumAsString();
+
+    public final String studentBlock;
+
+    public enum HallBlock {
+        A,
+        B,
+        C,
+        D,
+        E
+    }
+
+    /**
+     * Constructs a {@code Block}.
+     *
+     * @param block A valid block.
+     */
+    public Block(String block) {
+        requireNonNull(block);
+        checkArgument(isValidBlock(block), MESSAGE_CONSTRAINTS);
+        studentBlock = block.toUpperCase();
+    }
+
+    /**
+     * Returns true if a given string is a valid block.
+     * @param test
+     * @return true if a given string is valid, false otherwise.
```

Comment:
Maybe the first letter in the word true can be capitalised to standardise with the other methods' Java Doc.

Reply:
Noted! I've made the appropriate changes 😄